### ISSUE TITLE
Spawning Improvement during Invasions and more.

### ISSUE THREAD

**Issue by SirAron111** [Corosauce/HostileWorlds_Invasions#9]
Would it be possible to have an option that mob ignore light levels during invasions when checking for a spawn able are? But not simply ignore it that would result in them spawning next to players in their bases. But rather just ignore it when they can find a place to spawn because of light and then only at a set range like 40 blocks away from the player this would prevent them to spawn directly in the palyers base but still spawn them outside so they can go to them. Because atm it feels like when every area I s lit up they don’t spawn as much as they should? Correct me if I am wrong pls.

On the other hand there are a few things regarding the way mobs try to get to the player that need a bit of tweaking I think. 

1.	When a mob starts mining a block it should not stop and change the block it wants to mine just because the player moved like one block from one chest to the other.

2.	Mobs shouldn’t mine randomly around when they can’t find a target rather have them try to move to the last known location of the player as the player mostly will be around there. This prevents the destruction of some blocks that don’t respawn properly like farmlands/fences/and crops etc (I know you are already working on an even better thing to prevent this with the black list feature I just thought this would add to the effectiveness of pathing.)

3.	Mobs still have a hard time dealing with players living in Underground or Sky bases. For Sky bases they simply can’t get up to them isn’t there a way to give them a dedicated pillar block? Like the repairing block just for mobs to walk on (well that would be a bit immersion breaking but maybe that works with dedicated cmod ai that carries them and can build with them like a corpse block XD) or maybe simply have them throw each other upwards to the player when they can’t pillar up on each other high enough as again once the player moves just little they mostly stop towering and fall apart so the one that was digging trough the sky base falls and well mostly dies never finishing to dig up.

4.	For Underground bases Mobs can dig downwards until they reach the players base that is not a problem again only if the player moves they stop digging and try to find another way in which leaves them constantly changing their digged block so they never really finish.

5.	Another problem with underground bases would be that if there is a cave underneath or in modpacks a big cave system the invasion mobs that spawn down there suffer the same problem as if they try to get to a sky base. If they could somehow properly get up there that would help. Same problem with tall buildings when the player is so far up that they simply can’t find a place to spawn.

So basically by improving how mobs spawn so they don’t get stuck with the worst spawn places in some tiny underground corner and maybe have them sometimes finish their task of mining (or maybe have them finish digging the path they planned to take completely) would already improve mobs getting to the player a lot.

PS: you can ignore it all if it’s not possible or would take too much effort just wanted to get the ideas out of my head.

**Comment by SirAron111** [comment]
And one more thing IS it possible to make Blocks from Chisel and bits since they can’t be mined simply phase trough? So that if a mob encounters a Block during an Invasion that it can’t Mine It simply Phases trough that block? Like ignoring collision because that’s a huge problem in many ways as people can simply make a one bit thick barrier or so to block your Invasion mobs completely or take one corner of a block out to make it unbreakable for Invasion mobs. Simply make Mobs walk trough anything they can’t mine if it’s in their pathing way (but have them prefer a way where they can mine that would solve all problems even with fences and the like)

**Comment by slarzyer** [comment]
i have played with this mod installed and several points to make about the random mining and non-red blocks both seem like better pathing could help or some way for them to see them as a pass-thru block with no need to mine it similar to the way some mods "glass" will let certain entities thru it...for example i have created background in one spot and it no where near where i hang out but they mine it anyway...or maybe if you make only mine while player is in "set radius" could help and until the"set radius" they do a search pattern...

**Comment by slarzyer** [comment]
i hit enter on mistake.....the "group mentality" similar to pig-men may help if one detects "player radius" then it "calls" the group and the continuing their "1st mining path" without switching in great idea!
 
flowing water(liquids) also is an issue for the "red-blocks" if u could make them block flowing liquid would help.

**Comment by SirAron111** [comment]
Also a way to make sky platform bases less op would be an option to force spawn them in lit up areas disregarding player distance so that if someone tries to make a sky base to escape instead of making a well defended ground base he will get at least some of the mobs spawned right next to him or have them fall from the sky above him. Sky bases are the biggest weakness of this mod and if you don’t want to limit your players creativity on building but keep the challenge up that’s hard to do. Does anyone have a better idea to bring at least some invasion mobs into sky bases? (Would also enable the use of this mod with void / sky factory like worlds)

**Comment by SirAron111** [comment]
Another Idea for sky bases I recently came up with. Adding a flyer AI (Same as there is a Miner Ai.) that simply notices ye player is unpathable to since no connection and to high above and then the Ai hocks in.
Being able to custom set certain mobs to be able to fly/float upwards, would be a neat thing no extra spawning rules no extra  light ignoring etc just render a little wings at the back (gimmick) and have them be able to float upwards like balloons until they are at the same height or slightly higher than the player and then they drop down onto their sky platform or base (maybe the fliers could carry/pickup a random mob that isn’t a flyer like a miner and they would make a neat combo like air assault) by Simply adding not to many of these to the custom invasions, Sky bases would still be a good idea but not ultra save.
Well or simply having them climb ladders would also be one way to solve this problem early until a player makes a teleported or an elevator to escape. Just addding this idea here in case I forget it.

Do take your time Coro RL is more important.
